packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -0,0 +1,93 @@
+/** @jsx withSlots */
+import {
+  applyTokenLayers,
+  buildUseTokens,
+  compressible,
+  mergeStyles,
+  Theme,
+  withSlots,
+  useSlot,
+  useTheme,
+  UseTokens,
+  borderStyles,
+} from '@fluentui-react-native/framework';
+import { ShadowProps, ShadowTokens } from './Shadow.types';
+import { View, ViewProps } from 'react-native';
+import { applyPropsToTokens } from '@fluentui-react-native/use-tokens';
+
+const useShadowTokens = buildUseTokens<ShadowTokens>((theme: Theme) => {
+  return {
+    color: 'black',
+    backgroundColor: theme.colors.background,
+    ambient: theme.effects.shadow4.ambient,
+    key: theme.effects.shadow4.key,
+    innerShadowSlot: View,
+    depth2: {
+      ambient: theme.effects.shadow2.ambient,
+      key: theme.effects.shadow2.key,
+    },
+    depth4: {
+      ambient: theme.effects.shadow4.ambient,
+      key: theme.effects.shadow4.key,
+    },
+    depth8: {
+      ambient: theme.effects.shadow8.ambient,
+      key: theme.effects.shadow8.key,
+    },
+    depth16: {
+      ambient: theme.effects.shadow16.ambient,
+      key: theme.effects.shadow16.key,
+    },
+    depth28: {
+      ambient: theme.effects.shadow28.ambient,
+      key: theme.effects.shadow28.key,
+    },
+    depth64: {
+      ambient: theme.effects.shadow64.ambient,
+      key: theme.effects.shadow64.key,
+    },
+  };
+}, 'Shadow');
+
+export const Shadow = compressible<ShadowProps, ShadowTokens>((props: ShadowProps, useTokens: UseTokens<ShadowTokens>) => {
+  const theme = useTheme();
+  let [tokens, cache] = useTokens(theme);
+  const { depth, padding, paddingHorizontal, style, ...rest } = props;
+  [tokens, cache] = applyTokenLayers(
+    tokens,
+    ['depth2', 'depth4', 'depth8', 'depth16', 'depth28', 'depth64'],
+    cache,
+    (layer) => layer === 'depth' + depth,
+  );
+  [tokens, cache] = applyPropsToTokens(props, tokens, cache, ['borderColor', 'borderRadius', 'borderStyle', 'borderWidth']);
+  const Root = useSlot<ViewProps>(View, {
+    ...rest,
+    style: mergeStyles(
+      {
+        backgroundColor: tokens.backgroundColor,
+        shadowColor: tokens.color,
+        ...borderStyles.from(tokens, theme),
+        ...tokens.key,
+      },
+      style,
+    ),
```

Comment:
The object identity of the style used in merge styles needs to be consistent, not just the values within. This can be done via either `useMemo` or the `cache` you have at that point.

`useMemo` is a per-component run to run cache. Meaning if component instance A renders a second time it will re-use the object if the passed in key values are the same. Usage would look like this:

```ts
const rootStyle = React.useMemo(() => ({
  backgroundColor: tokens.backgroundColor,
  shadowColor: tokens.color,
  ...borderStyles.from(tokens, theme),
  ...tokens.key
}), [tokens, theme]);
```

The `cache` splits itself based on layer permutations and theme automatically so it is ready to use at this point. It stores styles across instances so multiple instances of `Shadow` with the same depth will end up using the same styles as long as the input parameters are the same. Usage of cache would look something like:

```ts
const [tokenStyles] = cache(() => ({
  rootStyle: {
    // same as above
  },
  innerStyle: {
    // second section below
  }),
  []); // no keys required since tokens + theme are already considered
```